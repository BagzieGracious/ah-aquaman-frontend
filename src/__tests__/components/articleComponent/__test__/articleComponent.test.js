import { shallow } from "enzyme";
import React from "react";
import ArticleComponent from "../../../../components/article/articleComponent";

describe("Article component test", () => {
  it("should render without crashing", () => {
    const props = {
      article: {
        title: "test title",
        image: "http://imageurl.com/article_image.jpg",
        tagList: [],
        author: {
          image: "http://imageurl.com/avatar.jpg",
          username: "testuser",
        },
      },
    };
    const wrapper = shallow(<ArticleComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Article component test", () => {
  it("should render without crashing", () => {
    const props = {
      article: {
        title: "test title",
        image: null,
        tagList: [],
        author: {
          image: "http://imageurl.com/avatar.jpg",
          username: "testuser",
        },
      },
    };
    const wrapper = shallow(<ArticleComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const props = {
      article: {
        title: "test title",
        image: null,
        tagList: [],
        read_time: "test read time",
        author: {
          image: "http://imageurl.com/avatar.jpg",
          username: "testuser",
        },
      },
    };
    const wrapper = shallow(<ArticleComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(props.article.read_time).toBe("test read time");
  });
});
