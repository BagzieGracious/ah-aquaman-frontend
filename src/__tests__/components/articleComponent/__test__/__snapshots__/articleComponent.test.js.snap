// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test should render without crashing 1`] = `ShallowWrapper {}`;

exports[`Article component test should render without crashing 2`] = `ShallowWrapper {}`;

exports[`Article component test should render without crashing 3`] = `ShallowWrapper {}`;
